### PR TITLE
Add enforcepgo for release Windows_NT default release build

### PR DESCRIPTION
**Pull request by nategraf**
This PR adds a step in the process for the windows release build scenario to enforce that PGO is used in compilation of coreclr.dll and clrjit.dll. The intention is to prevent merging any code which would break PGO

This is my first time looking at this system, so I'd like to confirm my change does what I expect. Mainly, what I am unsure of is whether or not I am correct to limit the additional of 'enforcepgo' to default scenario builds, or if I should add it to all release builds.

@jashook @mmitche I added you two because you are top contributors on this file recently, let me know if there is someone else I should ask to review this change instead

Follow-up to #13258 
Connect #12863

### CODE REVIEW

**Comment by brianrob** [comment]
@nategraf, this looks good from a "we want to do this" perspective.  I will defer to @mmitche and @jashook on the implementation itself.

@adiaaida may also be able to review this.

File: netci.groovy [hunk 1]
```diff
@@ -1442,6 +1442,11 @@ def static calculateBuildCommands(def newJob, def scenario, def branch, def isPR
                         buildCommands += "set __TestIntermediateDir=int&&build.cmd ${lowerConfiguration} ${arch} ${buildOpts}"
                     }
 
+                    // If it is a release build for windows, ensure PGO is used, else fail the build
```
Comment (jashook) [inline netci.groovy]: Looks like this change will get picked up for x86 Legacy Backend and x86 Ryujit. I am assuming x64 and x86 Ryujit are the only interesting cases

Comment (adiaaida) [inline netci.groovy]: Agreed that we shouldn't enforce pgo for legacy backend, since we aren't collecting for it.

Comment (nategraf) [inline netci.groovy]: Let me know if this commit doesn't fix that 8cff14e28ebf2aaa31956962231a54bdab1974a6

**Comment by nategraf** [comment]
I've expanded the use of the `enforcepgo` flag in ff4553a to all the basic scenarios after I realised that there is no `default` release scenario run on GirHub PRs automatically.

@jashook do you see any problem with this?

**Comment by jashook** [comment]
@dotnet-bot test CI Please

**Comment by nategraf** [comment]
@dotnet-bot test CI please

**Comment by jkotas** [comment]
@nategraf All Windows Release jobs started failing with:

```
14:00:18 Error: Parameter not recognized: release.
14:00:18 Our dev workflow has changed! Use -? for help in the new options we have and how to pass parameters now.
```

It seems to be caused by this change.

**Comment by nategraf** [comment]
I have been looking into this for the past hour and just figured it out. There is an issue with the way whitespace is parsed at the command line https://github.com/dotnet/coreclr/issues/13317

**Pull request merged**